client/my-sites/plugins/disconnect-jetpack/disconnect-jetpack-dialog.jsx

```diff
@@ -39,14 +39,15 @@ module.exports = React.createClass( {
 	},
 
 	disconnectJetpack: function() {
-		var jetpackSites = sites.getJetpack();
+		var jetpackSites = sites.getJetpack(),
+			selectedSite = sites.getSelectedSite();
 
 		// remove any error and completed notices
 		SitesListActions.removeSitesNotices( [ { status: 'error' }, { status: 'completed' } ] );
 
 		if ( this.props.site ) {
 			SitesListActions.disconnect( this.props.site );
-			if ( this.props.redirect && jetpackSites.some( this.jetpackSiteRemain ) ) {
+			if ( selectedSite === this.props.site && this.props.redirect && jetpackSites.some( this.jetpackSiteRemain ) ) {
```

Comment:
I don't understand it 100% ... but know I'm looking at it I think we don't need it anymore ( it only should go into this condition if the selected site is the one being deleted, and in this case jetpackSiteRemain() return will always be a jetpack site, so ... we can remove that part of the condition without side effects?